v2/tools/generator/internal/codegen/pipeline/repair_skipping_properties.go

```diff
@@ -43,47 +43,49 @@ const DetectSkippingPropertiesStageID = "detectSkippingProperties"
 // KnownAs: also present in (v2, v3); no issue
 // GivenName: present in (v1); no issue
 //
-// Address: present in (v1, v3); (skipping v2) issue to report, aborting the pipeline.
-//
-// Additional complexities:
-// - We need to handle type renaming between versions.
-// - When introduced, we will also need to handle property renaming between versions
-func DetectSkippingProperties() *Stage {
+// Address: present in (v1, v3); (skipping v2), repair required.
+func RepairSkippingProperties() *Stage {
 	return NewStage(
-		DetectSkippingPropertiesStageID,
-		"Detect properties that skip resource or object versions",
+		RepairSkippingPropertiesStageID,
+		"Repair property bag serialization for properties that skip resource or object versions",
 		func(ctx context.Context, state *State) (*State, error) {
-			detector := newSkippingPropertyDetector(state.Definitions(), state.ConversionGraph())
+			repairer := newSkippingPropertyRepairer(state.Definitions(), state.ConversionGraph())
 
 			// Add resources and objects to the graph
 			for _, def := range state.Definitions() {
 				if t, ok := astmodel.AsPropertyContainer(def.Type()); ok {
-					err := detector.AddProperties(def.Name(), t.Properties().AsSlice()...)
+					err := repairer.AddProperties(def.Name(), t.Properties().AsSlice()...)
 					if err != nil {
 						return nil, err
 					}
 				}
 			}
 
-			err := detector.CheckForSkippedProperties()
+			defs, err := repairer.RepairSkippedProperties()
+			if err != nil {
+				return nil, err
+			}
 
-			return state, err
+			return state.WithOverlaidDefinitions(defs), err
```

Comment:
Do we need pre/post requisite check for `CreateConversionGraph` stage here?

Reply:
Our checks can't handle requiring the same stage both before and after; I've added a prerequisite check.